Core/Conditions: Added condition for nearby personal-spawn creature

matanshukry opened the pull request:
**Changes proposed:**

-  Added a condition to check for a nearby creature that is a personal spawn

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**

**Tests performed:**

(Does it build, tested in-game, etc.)

- [x] It builds
- [x] Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

matanshukry committed:
Added condition to check for personal creature nearby

Faq commented on src/server/game/Conditions/ConditionMgr.h:
```diff
@@ -106,6 +106,7 @@ enum ConditionTypes
     CONDITION_DIFFICULTY_ID            = 49,                   // Difficulty       0              0                  true is map has difficulty id
     CONDITION_OBJECT_ENTRY_GUID        = 51,                   // TypeID           entry          guid               true if object is type TypeID and the entry is 0 or matches entry of the object or matches guid of the object
     CONDITION_TYPE_MASK                = 52,                   // TypeMask         0              0                  true if object is type object's TypeMask matches provided TypeMask
+    CONDITION_NEAR_PERSONAL_CREATURE   = 53,                   // creature entry   distance       dead (0/1)         true if there is a personal creature of entry in range
```
```suggestion
    CONDITION_NEAR_PERSONAL_CREATURE   = 53,                   // creature entry   distance       dead (0/1)         true if there is a personal-spawn creature of entry in range
```

matanshukry committed:
Update src/server/game/Conditions/ConditionMgr.h

Co-authored-by: NoName <322016+Faq@users.noreply.github.com>

Shauren commented on src/server/game/Grids/Notifiers/GridNotifiers.h:
```diff
@@ -1307,12 +1307,13 @@ namespace Trinity
             {
-                if (u->getDeathState() != DEAD && u->GetEntry() == i_entry && u->IsAlive() == i_alive && i_obj.IsWithinDistInMap(u, i_range))
+                if (u->getDeathState() != DEAD && u->GetEntry() == i_entry && u->IsAlive() == i_alive && i_obj.IsWithinDistInMap(u, i_range) &&
+                    (i_personalSpawnOwner == nullptr || u->GetOwner() == i_personalSpawnOwner))
```
Personal spawn uses different logic - https://github.com/TrinityCore/TrinityCore/blob/e0278f05379ed470be8264d3f362360a52b3454d/src/server/game/Entities/Object/Object.cpp#L1450-L1452

matanshukry committed:
- Using WorldObject rather than Unit for owner (as that is the base)
- Checking for personal spawn correctly in NearestCreatureEntryWithLiveStateInObjectRangeCheck rather than simply checking for owner

matanshukry committed:
Merge branch 'issue/find-near-personal-creature' of github.com:matanshukry/TrinityCore into issue/find-near-personal-creature

Shauren commented:
Replaced by #26102
